lib/rubocop/cop/rspec/implicit_block_expectation.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Check that implicit block expectation syntax is not used.
+      #
+      # Prefer using explicit block expectations.
+      #
+      # @example
+      #   # bad
+      #   subject { -> { do_something } }
+      #   it { is_expected.to change(something).to(new_value) }
+      #
+      #   # good
+      #   it 'changes something to a new value' do
+      #     expect { do_something }.to change(something).to(new_value)
+      #   end
+      class ImplicitBlockExpectation < Cop
+        MSG = 'Avoid implicit block expectations.'
+
+        def_node_matcher :lambda?, <<-PATTERN
+          {
+            (send (const nil? :Proc) :new)
+            (send nil? :proc)
+            (send nil? :lambda)
+          }
+        PATTERN
+
+        def_node_matcher :lambda_subject?, <<-PATTERN
+          (block _ _args $(block #lambda? ...))
+        PATTERN
+
+        def_node_matcher :implicit_expect, <<-PATTERN
```

Comment:
Shamelessly borrowed from `ImplicitSubject`. Not sure if it makes sense to extract this with just two usages.